### ISSUE TITLE
--extra_vars not working with var in playbook

### ISSUE THREAD

**Issue by mattwillsher** [ansible/ansible#1695]
http://ansible.cc/docs/playbooks2.html#passing-variables-on-the-command-line isn't working.

With hosts as a variable:

```
$ cat inventory

[all]
a
b
c

$ cat release.yml

---
 - hosts: $hosts
   user: root

$ ansible-playbook -i inventory -e hosts=all release.yml --list-hosts
playbook: release.yml
  hosts in play 1 (unnamed): #0
```

With hosts statically assigned:

```
$ cat release.yml

---
 - hosts: all
   user: root

$ ansible-playbook -i inventory release.yml --list-hosts
playbook: release.yml
  hosts in play 1 (unnamed): #3
    a
    b
    c
```

I'd expect both outputs to be the same.

**Comment by dhozac** [comment]
--list-hosts is a very special case that doesn't use most of the code that the rest of ansible relies on. Should be an easy fix though...

**Comment by dhozac** [comment]
Fixed by #1753.
